pixels-turbo/pixels-worker-vhive/src/main/java/io/pixelsdb/pixels/worker/vhive/PixelsWriterStreamImpl.java

```diff
@@ -0,0 +1,830 @@
+import io.pixelsdb.pixels.common.utils.ConfigFactory;
+import io.pixelsdb.pixels.common.utils.Constants;
+import io.pixelsdb.pixels.core.PixelsProto;
+import io.pixelsdb.pixels.core.PixelsWriter;
+import io.pixelsdb.pixels.core.PixelsWriterImpl;
+import io.pixelsdb.pixels.core.TypeDescription;
+import io.pixelsdb.pixels.core.encoding.EncodingLevel;
+import io.pixelsdb.pixels.core.exception.PixelsWriterException;
+import io.pixelsdb.pixels.core.vector.ColumnVector;
+import io.pixelsdb.pixels.core.vector.VectorizedRowBatch;
+import io.pixelsdb.pixels.core.writer.ColumnWriter;
+import io.pixelsdb.pixels.core.writer.PixelsWriterOption;
+import io.pixelsdb.pixels.turbo.StreamProto;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.asynchttpclient.*;
+import org.springframework.retry.support.RetryTemplate;
+
+import javax.annotation.concurrent.NotThreadSafe;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.ByteOrder;
+import java.util.List;
+import java.util.Optional;
+import java.util.TimeZone;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.netty.handler.codec.http.HttpHeaderNames.*;
+import static io.netty.handler.codec.http.HttpHeaderValues.CLOSE;
+import static io.pixelsdb.pixels.common.utils.Constants.MAGIC;
+import static io.pixelsdb.pixels.core.writer.ColumnWriter.newColumnWriter;
+import static io.pixelsdb.pixels.worker.vhive.StreamWorkerCommon.getPort;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * PixelsWriterStreamImpl is an implementation of {@link PixelsWriter} that writes
+ * ColumnChunks to a stream, for operator pipelining over HTTP.
+ *
+ * <p>
+ * DESIGN:
+ * In partitioned mode, each partition worker divides its assigned file list into multiple partitions.
+ * Each partition contains several files identified by a partitionId, corresponding to the partition worker's workerId:
+ *   0   1   2   3   4
+ * Within each partition, data is further hashed based on a hash value that corresponds to the workerId of the next-level workers:
+ * ---------------------
+ * | 0 | 0 | 0 | 0 | 0 |
+ * |---|---|---|---|---|
+ * | 1 | 1 | 1 | 1 | 1 |
+ * |---|---|---|---|---|
+ * | 2 | 2 | 2 | 2 | 2 |
+ * |---|---|---|---|---|
+ *
+ * Each partition worker sends its hashed data parts to the corresponding join workers in sequence. For example:
+ *  - Partition worker 0 sends its hash=0 part (of partition 0) to join worker 0, hash=1 part to join worker 1, and so on.
+ *  - The same pattern is followed by partition workers 1, 2, 3, 4, etc.
+ * Each join worker listens on a specific port for all parts with the same hash value across all partitions.
+ * Consequently, a partition worker must send each hash part (within its partition) to different ports.
+ */
+@NotThreadSafe
+public class PixelsWriterStreamImpl implements PixelsWriter {
```

Comment:
Please reformat the code following the regular code style in pixels.
For class, method, and static code block, loop, if else, and try catch, we use aligned {} pair:
```java
class A
{
    private void func ()
    {
        try (Connection conn....)
        {
            for (int i : list)
            {
                if (i > 2)
                {
                } else
                {
                }
            }
        } catch (Exception e)
        {
        }
    }

    static
    {
    }
}
```

And it is recommended that no line is longer than 120 characters.